lib/mix/lib/mix/tasks/xref.ex

```diff
@@ -283,6 +320,85 @@ defmodule Mix.Tasks.Xref do
     Mix.raise message
   end
 
+  ## Graph helpers
+
+  defp excluded(opts) do
+    Keyword.get_values(opts, :exclude)
+    |> Enum.map(&{&1, nil})
+  end
+
+  defp module_references() do
+    source_modules =
+      for manifest <- E.manifests(),
+          manifest_data = read_manifest(manifest, ""),
+          module(module: module, source: source) <- manifest_data,
+          source = Enum.find(manifest_data, &match?(source(source: ^source), &1)),
+          do: {source, module}
+
+    source_modules = Enum.group_by(source_modules, &elem(&1, 0), &elem(&1, 1))
+
+    all_modules =
+      Enum.reduce source_modules, MapSet.new(), fn {_, modules}, acc ->
+        Enum.reduce(modules, acc, &MapSet.put(&2, &1))
+      end
+
+    Enum.reduce source_modules, [], fn {source, modules}, acc ->
+      source(runtime_references: r, compile_references: c) = source
+      references =
+        r ++ c ++ modules
+        |> MapSet.new()
+        |> MapSet.intersection(all_modules)
+
+      Enum.reduce modules, acc, fn module, acc ->
+        module_references = MapSet.delete(references, module)
+        [{{inspect(module), nil}, Enum.map(module_references, &{inspect(&1), nil})} | acc]
+      end
+    end
+  end
+
+  defp write_graph(module_references, excluded, opts) do
+    {root, module_references} =
+      case {opts[:source], opts[:sink]} do
```

Comment:
So where do you remove the references that are not even indirectly linked
to the source/sink?

On Monday, June 13, 2016, Eric Entin <notifications@github.com> wrote:

> In lib/mix/lib/mix/tasks/xref.ex
> <https://github.com/elixir-lang/elixir/pull/4811#discussion_r66791201>:
>
> > +      source(runtime_references: r, compile_references: c) = source
> > +      references =
> > +        r ++ c ++ modules
> > +        |> MapSet.new()
> > +        |> MapSet.intersection(all_modules)
> > +
> > +      Enum.reduce modules, acc, fn module, acc ->
> > +        module_references = MapSet.delete(references, module)
> > +        [{{inspect(module), nil}, Enum.map(module_references, &{inspect(&1), nil})} | acc]
> > +      end
> > +    end
> > +  end
> > +
> > +  defp write_graph(module_references, excluded, opts) do
> > +    {root, module_references} =
> > +      case {opts[:source], opts[:sink]} do
>
> That's what this code does, though?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/elixir-lang/elixir/pull/4811/files/75f10cd6777df30062a09f69a4c46764739f261a#r66791201>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAAlbtrJCck1_32f0NON3g9oDOzk1hJeks5qLVuZgaJpZM4Iz_6f>
> .
>


-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*